pkg/utils/store.go

```diff
@@ -78,7 +79,7 @@ func StorageProviderFromInfraProvider(infra *druidv1alpha1.StorageProvider) (str
 	}
 
 	switch *infra {
-	case aws, S3:
+	case aws, stackit, S3:
 		return S3, nil
 	case azure, ABS:
 		return ABS, nil
```

Comment:
Could you add another `case` to the `switch` instead of adding it to the `aws, S3` case, since it would make it cleaner for all S3 compatible providers to be in their own case sepearate from AWS S3?
Not a strong opinion whatsoever, just a style preference. Totally up to you.
```suggestion
	switch *infra {
	case aws, S3:
		return S3, nil
	// S3 compatible providers
	case stackit:
        return S3, nil
	case azure, ABS:
		return ABS, nil
```